internal/hvsocket/hvsocket.go

```diff
@@ -0,0 +1,91 @@
+//go:build windows
+// +build windows
+
+package hvsocket
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+	"strings"
+	"unicode/utf16"
+	"unsafe"
+
+	"github.com/Microsoft/go-winio/pkg/guid"
+	"golang.org/x/sys/windows"
+)
+
+func HCSIDToGUID(id string) (guid.GUID, error) {
+	var buf bytes.Buffer
+	if err := binary.Write(&buf, binary.LittleEndian, utf16.Encode([]rune(strings.ToUpper(id)))); err != nil {
+		return guid.GUID{}, err
+	}
+	g, err := guid.NewV5(guid.GUID{Data1: 0xcab70344, Data2: 0xfacb, Data3: 0x41e4, Data4: [8]byte{0xb5, 0xe5, 0xab, 0x65, 0x92, 0x28, 0x3e, 0x6e}}, buf.Bytes())
+	if err != nil {
+		return guid.GUID{}, err
+	}
+	return g, nil
+}
+
+const (
+	cAddressFlagPassthru            = 0x00000001
+	cIOCtlHVSocketUpdateAddressInfo = 0x21c004
+)
+
+type tAddressInfo struct {
+	SystemID         guid.GUID
+	VirtualMachineID guid.GUID
+	SiloID           guid.GUID
+	Flags            uint32
+}
+
+type Handle windows.Handle
+
+func (h Handle) Release(ctx context.Context) error {
+	return windows.CloseHandle(windows.Handle(h))
+}
+
+func CreateAddressInfo(cid, vmid guid.GUID, passthru bool) (Handle, error) {
```

Comment:
it might be better to keep the type private and also make it a proper struct to prevent casting from handles not created below:

```suggestion
type addressInfo {
	h windows.Handle
}

var _ resources.ResourceCloser = addressInfo{}

func (ai addressInfo) Release(ctx context.Context) error {
	return windows.CloseHandle(windows.Handle(ai.h))
}

func CreateAddressInfo(cid, vmid guid.GUID, passthru bool) (resources.ResourceCloser , error) {
```